crates/circuit/src/dag_circuit.rs

```diff
@@ -5843,12 +5830,16 @@ impl DAGCircuit {
                 )?
                 .into_any()
             }
-            NodeType::VarIn(var) => {
-                Py::new(py, DAGInNode::new(py, id, var.clone_ref(py)))?.into_any()
-            }
-            NodeType::VarOut(var) => {
-                Py::new(py, DAGOutNode::new(py, id, var.clone_ref(py)))?.into_any()
-            }
+            NodeType::VarIn(var) => Py::new(
+                py,
+                DAGInNode::new(py, id, self.vars.get(*var).unwrap().clone_ref(py)),
+            )?
+            .into_any(),
+            NodeType::VarOut(var) => Py::new(
+                py,
+                DAGOutNode::new(py, id, self.vars.get(*var).unwrap().clone_ref(py)),
+            )?
+            .into_any(),
```

Comment:
Bit of a nitpick since from what it seems this would have no big effects but you could rewrite it this way (which I alluded to before):
```suggestion
            NodeType::VarIn(var) => DAGInNode::new(py, id, self.vars.get(*var).unwrap().clone_ref(py)).into_py(py),
            NodeType::VarOut(var) => DAGOutNode::new(py, id, self.vars.get(*var).unwrap().clone_ref(py)).into_py(py),
```

It might be too long for a one-liner though 😅

Reply:
I rather like that approach! Though, IMO we ought to change over to using it consistently in a separate PR (if we feel the improved readability is worth the effort).

The reason being that it keeps the diff for this PR's commit onto `main` as logically specific / isolated as possible. This way, if there's an issue in the future (or someone wants to see exactly the changes that were strictly necessary to do XYZ), there isn't any extra fluff to sift through when they're looking through the Git history.

I'd be more inclined to use the approach you mention despite this if there weren't other uses of `Py::new` inside this very method, but since there are I'd be touching lines that do not require modifications to achieve what this PR aims to do.

Reply:
Alright, then, this looks good either way, good job!